src/permissionSelectors.js

```diff
@@ -40,6 +42,105 @@ export function roleIsAtLeast(thisRole: RoleT, thresholdRole: RoleT): boolean {
   return thisRole <= thresholdRole; // Roles with more privilege have lower numbers.
 }
 
+/**
+ * Whether the user has passed the realm's waiting period to be a full member.
+ *
+ * See:
+ *   https://zulip.com/api/roles-and-permissions#determining-if-a-user-is-a-full-member
+ *
+ * To determine if a user is a full member, callers must also check that the
+ * user's role is at least Role.Member.
+ *
+ * Note: If used with useSelector, the value will be out-of-date, though
+ * realistically only by seconds or minutes at most; see implementation.
+ */
+export function getHasUserPassedWaitingPeriod(state: PerAccountState, userId: UserId): boolean {
+  const { waitingPeriodThreshold } = getRealm(state);
+  const { date_joined } = getUserForId(state, userId);
+
+  return (
+    differenceInDays(
+      // When used with useSelector, the result will be based on the time at
+      // which the most recent Redux action was dispatched. This would break
+      // if we made this a caching selector; don't do that.
+      // TODO(?): To upper-bound how long ago that can be, we could dispatch
+      //   actions at a regular short-ish interval. If those actions
+      //   contained a current-date value, we could even store that value in
+      //   Redux and consume it here, letting this be a caching selector if
+      //   we wanted it to be.
+      new Date(),
+      // The server doc doesn't give a format for date_joined, but it'd be
+      // surprising if the Date constructor couldn't handle it.
+      new Date(date_joined),
```

Comment:
Hmm, yeah, that's annoying.

There's a clue in the example responses, though, on endpoints like https://zulip.com/api/get-own-user :

    "date_joined": "2019-10-20T07:50:53.728864+00:00",

So it looks like it's in ISO 8601 format: https://en.wikipedia.org/wiki/ISO_8601 A good choice for an API.

And it [looks like](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#date_time_string_format) `new Date` is specified to interpret those correctly. There's advice not to use it because it also recognizes implementation-specific formats, but as long as we stick to the standard format that seems like not an issue.